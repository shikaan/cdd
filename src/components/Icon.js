import React from "react";
import styled from "styled-components";

const IconWrapper = styled.i`
  font-style: normal;
`

export default class Icon extends React.Component {
  render() {
    const {icon, className, children} = this.props;

    const iconClassName = `cdd-${icon || String(children)}`
    const classes = className
      ? `${className} ${iconClassName}`
      : iconClassName

    return <IconWrapper className={classes}/>;
  }
}
