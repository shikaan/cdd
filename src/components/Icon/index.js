import React from "react";
import styled from "styled-components";

const StyledI = styled.i({
  fontStyle: "normal"
});

export default class Icon extends React.Component {
  render() {
    const {icon, className, children} = this.props;

    const iconClassName = `cdd-${icon || String(children)}`
    const classes = className
      ? `${className} ${iconClassName}`
      : iconClassName

    return <StyledI className={classes}/>;
  }
}
