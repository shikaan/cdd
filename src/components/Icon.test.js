import React from "react";
import {shallow} from "enzyme";
import "jest-styled-components";

import Icon from "./Icon.js";

const fixutres = {
  icon: "icon",
  className: "className"
};

describe("Icon", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Icon icon={fixutres.icon}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it("adds the class matching provided icon", () => {
    const wrapper = shallow(<Icon icon={fixutres.icon}/>);

    expect(wrapper.exists(`.fswb-${fixutres.icon}`)).toBeTruthy();
  });

  it("allows other classes along with icon", () => {
    const wrapper = shallow(<Icon className={fixutres.icon} icon={fixutres.icon}/>);

    expect(wrapper.exists(`.${fixutres.icon}.fswb-${fixutres.icon}`)).toBeTruthy();
  });
});
