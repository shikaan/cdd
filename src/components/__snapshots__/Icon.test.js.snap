// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<styled.i
  className="fswb-icon"
/>
`;
